sktime/transformations/multiplexer.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements transformer for selecting among different model classes."""
+
+from sktime.base import _HeterogenousMetaEstimator
+from sktime.transformations._delegate import _DelegatedTransformer
+from sktime.transformations.base import BaseTransformer
+
+__author__ = ["miraep8"]
+__all__ = ["MultiplexTransformer"]
+
+
+class MultiplexTransformer(_DelegatedTransformer, _HeterogenousMetaEstimator):
+    """Facilitate an AutoML based selection of the best transformer.
+
+    When used in combination with either TransformedTargetForecaster or
+    ForecastingPipeline in combination with ForecastingGridSearchCV
+    MultiplexTransformer provides a framework for transformer selection.  Through
+    selection of the appropriate pipeline (ie TransformedTargetForecaster vs
+    ForecastingPipeline) the transformers in MultiplexTransformer will either be
+    applied to exogenous data, or to the target data.
+
+    MultiplexTransformer delegates all transforming tasks (ie, calls to fit, transform,
+    inverse_transform, and update) to a copy of the transformer in transformers
+    whose name matches selected_transformer.  All other transformers in transformers
+    will be ignored.
+
+    Parameters
+    ----------
+    transformers : list of sktime transformers, or
+        list of tuples (str, estimator) of named sktime transformers
+        MultiplexTransformer can switch ("multiplex") between these transformers.
+        Note - all the transformers passed in "transformers" should be thought of as
+        blueprints.  Calling transformation functions on MultiplexTransformer will not
+        change their state at all. - Rather a copy of each is created and this is what
+        is updated.
+    selected_transformer: str or None, optional, Default=None.
+        If str, must be one of the transformer names.
+            If passed in transformers were unnamed then selected_transformer must
+            coincide with auto-generated name strings.
+            To inspect auto-generated name strings, call get_params.
+        If None, selected_transformer defaults to the name of the first transformer
+           in transformers.
+        selected_transformer represents the name of the transformer MultiplexTransformer
+           should behave as (ie delegate all relevant transformation functionality to)
+
+    Attributes
+    ----------
+    transformer_ : sktime transformer
+        clone of the transformer named by selected_transformer to which all the
+        transformation functionality is delegated to.
+    _transformers : copy of the list of transformers passed.  If transformers was
```

Comment:
I think these are not copies but direct references

Reply:
Yes, I can see how the language would be confusing.